backend/src/main/java/com/redhat/cloud/notifications/routers/internal/HeapDumpResource.java

```diff
@@ -0,0 +1,63 @@
+package com.redhat.cloud.notifications.routers.internal;
+
+import com.sun.management.HotSpotDiagnosticMXBean;
+import io.quarkus.logging.Log;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.Response;
+import org.apache.http.HttpStatus;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import javax.management.MBeanServer;
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+
+import static com.redhat.cloud.notifications.Constants.API_INTERNAL;
+import static com.redhat.cloud.notifications.auth.ConsoleIdentityProvider.RBAC_INTERNAL_ADMIN;
+import static jakarta.ws.rs.core.MediaType.APPLICATION_OCTET_STREAM;
+
+@Path(API_INTERNAL + "/heap_dump")
+@RolesAllowed(RBAC_INTERNAL_ADMIN)
+public class HeapDumpResource {
+
+    @ConfigProperty(name = "quarkus.log.cloudwatch.log-stream-name", defaultValue = "localhost")
+    String hostname;
+
+    @GET
+    @Path("{hostname}")
+    @Produces(APPLICATION_OCTET_STREAM)
+    public Response getHeapDump(@PathParam("hostname") String hostname) {
+        if (!this.hostname.equals(hostname)) {
+            Log.infof("%s doesn't match with this host (%s)", hostname, this.hostname);
+            return Response.status(HttpStatus.SC_SEE_OTHER).build();
+        }
+
+        String tmpFileName = "notifications_dump.hprof";
+        String fullPathTmpFile = String.format("%s/%s", System.getProperty("java.io.tmpdir"), tmpFileName);
```

Comment:
We could do this the [clowder-quarkus-config-source](https://github.com/RedHatInsights/clowder-quarkus-config-source/blob/498a0b1ebac6ac5a6524892ef09b3ffe1a9029d1/src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java#L645-L653) way:
```java
File file = File.createTempFile("notifications_dump", ".hprof");
try {
    file.deleteOnExit();
} catch (SecurityException e) {
    log.warn("Delete on exit of the notifications heap dump file denied by the security manager", e);
}
```

Reply:
`deleteOnExit()` will delete file when the application will terminate, not only the method.
The idea is to be able to generate sevral heap dumps to check memory increasing before backends reboots.